src/manage_sql_secinfo.c

```diff
@@ -2081,14 +2242,435 @@ insert_scap_cpe_details (inserts_t *inserts, element_t cpe_item)
+  if (name == NULL)
+    {
+      g_warning ("%s: 'cpeName' field missing or not a string", __func__);
+      return -1;
+    }
+
+  refs = cJSON_GetObjectItemCaseSensitive (cpe_item, "refs");
+  if (! cJSON_IsArray (refs))
+    {
+      g_debug ("%s: 'refs' field missing or not an array", __func__);
+      return 0;
+    }
+
+  quoted_name = fs_to_uri_convert_and_quote_cpe_name (name);
+  cJSON_ArrayForEach (refs_item, refs)
+    {
+      int first;
+      char *ref, *type;
+      gchar *quoted_ref, *quoted_type;
+      ref = json_object_item_string (refs_item, "ref");
+      if (ref == NULL)
+        {
+          g_warning ("%s: 'ref' field missing or not a string", __func__);
+          g_free (quoted_name);
+          return -1;
+        }
+      type = json_object_item_string (refs_item, "type");
+      quoted_ref = sql_quote (ref ? ref : "");
+      quoted_type = sql_quote (type ? type : "");
+      
+      first = inserts_check_size (inserts);
+
+      g_string_append_printf (inserts->statement,
+                              "%s ('%s', '%s', '%s')",
+                              first ? "" : ",",
+                              quoted_name,
+                              quoted_ref,
+                              quoted_type);
+
+      inserts->current_chunk_size++;
+    }
+  g_free (quoted_name);
+
+  return 0;
+}
+
+/**
+ * @brief Update SCAP CPEs from a JSON file.
+ *
+ * @param[in]  path             Path to file.
+ *
+ * @return 0 success, -1 error.
+ */
+static int
+update_scap_cpes_from_json_file (const gchar *path)
+{
+  inserts_t inserts, deprecated_by_inserts;
+  gvm_json_pull_parser_t parser;
+  gvm_json_pull_event_t event;
+  FILE *json_stream = fopen (path, "r");
+  if (json_stream == NULL)
+    {
+      g_warning ("%s: Could not open file '%s': %s",
+                 __func__, path, strerror(errno));
+      return -1;
+    }
+
+  gvm_json_pull_parser_init (&parser, json_stream);
+  gvm_json_pull_event_init (&event);
+  if (scap_cpes_json_skip_to_products (&parser, &event))
+    {
+      gvm_json_pull_event_cleanup (&event);
+      gvm_json_pull_parser_cleanup (&parser);
+      fclose (json_stream);
+      return -1;
+    }
+
+  sql_begin_immediate ();
+  inserts_init (&inserts,
+                CPE_MAX_CHUNK_SIZE,
+                setting_secinfo_sql_buffer_threshold_bytes (),
+                "INSERT INTO scap2.cpes"
+                " (uuid, name, title, creation_time,"
+                "  modification_time, deprecated,"
+                "  cpe_name_id)"
+                " VALUES",
+                " ON CONFLICT (uuid) DO UPDATE"
+                " SET name = EXCLUDED.name,"
+                "     title = EXCLUDED.title,"
+                "     creation_time = EXCLUDED.creation_time,"
+                "     modification_time = EXCLUDED.modification_time,"
+                "     deprecated = EXCLUDED.deprecated,"
+                "     deprecated_by_id = EXCLUDED.deprecated_by_id,"
```

Comment:
I think this one can be removed now.
```suggestion
                "     deprecated_by_id = EXCLUDED.deprecated_by_id,"
```